### PR TITLE
Allow to edit admin/own user in the user management

### LINKED ISSUE #18530: Editing my user via user managment not possible
### Steps to reproduce
1. Login as Admin
2. Go to user management
3. Try to change my quota

### Expected behaviour

I can change the quota.

### Actual behaviour

I can not change to quota for admin. It's possible to change the quota for other users.

Peek 2019-12-22 12-42

cc @nextcloud/designers

### LINKED ISSUE #18964: Current user cannot edit groups (18RC2)
### Steps to reproduce
1. Open user management, all users except logged-in user have "Edit"-pencil icons
2. Try to add the current (in my case admin) user to another group

### Expected behaviour
As logged-in (admin) user, I should be able to add myself to extra groups.

### Actual behaviour
As logged-in (admin) user, I cannot edit my group memberships.

### relevant Server configuration
**Nextcloud version:** 18.0.0 RC2, updated from 17.0.2, originally installed from nextcloud.com/install
**Signing status:**: No errors have been found.

### LINKED ISSUE #19004: NC18 - can not change own passwd / group
hi,
i have made fresh install of nextcloud 18 on my own debian 10 server.
i can not change my own groups / passwd etc. if i log in as user1 (admin)
so i create a new user (user2) with admin privilegs and log in as user2
now i can not change user2 own groups / passwd etc...
but from user1 i can change all now
so login as user1 again i can change groups + passwd only from user2
but not my own (user1)...
this must be a bug ?

nextcloud/index.php/settings/users

### LINKED ISSUE #19072: Can not change admin settings
Good morning, when i go to the settings users, i can modify all settings but not mine, the admin user.

So i can't join any group, change my quota etc.

Nextcloud : v18

### Steps to reproduce
1. Go to settings

### Expected behaviour
Change admin (my) settings

### Actual behaviour
Link for change settings is not showing

### Server configuration

**Operating system:**
Nextcloud docker V18

### PR DESCRIPTION
**Pull request by juliusknorr**
- Fix own user editing (Fixes #18530, Fixes #18964, Fixes #19004, Fixes #19072) 
- Fix height of user rows 
- Fix missing submit button in the display name field
- Fix group navigation (regression from #18917) **No backport needed for this**

### CODE REVIEW

File: apps/settings/src/components/UserList/UserRowSimple.vue [hunk 1]
```diff
@@ -142,10 +141,6 @@ export default {
 			}
 			return t('settings', '{size} used', { size: OC.Util.humanFileSize(0) })
 		},
-		canEdit() {
-			return getCurrentUser().uid !== this.user.id && this.user.id !== 'admin'
```
Comment (skjnldsv) [inline apps/settings/src/components/UserList/UserRowSimple.vue]: I actually think it was something for the subadmins here.
There is indeed an issue, but removing the check might not be the solution.
Subadmins should not be able to edit themselves nor edit the admin user iirc :)

Comment (kesselb) [reply apps/settings/src/components/UserList/UserRowSimple.vue]: `this.user.id !== 'admin'` seems to be wrong anyway. I can pick any name for my admin user :thinking:

Comment (skjnldsv) [reply apps/settings/src/components/UserList/UserRowSimple.vue]: There is definitely something to change here, I agree, but I think this was here (not exactly like that before the multi line pr), that was needed before?

Comment (juliushaertl) [reply apps/settings/src/components/UserList/UserRowSimple.vue]: I don't see an issue with that even with those lines removed. 

user1 is group admin of group1
user2 is member of group1
admin is member of group1

As in the following screenshot (from user1) admin cannot be edited and user1 is not shown in general: 

image

Comment (skjnldsv) [reply apps/settings/src/components/UserList/UserRowSimple.vue]: Then all good! :)

**Review by skjnldsv — APPROVED** [review]

**Comment by kesselb** [comment]
Yes. I can edit myself again :+1: 

1: Login as admin and check the userlist.

image

2: Login as alice (subadmin of testgroup) and check the userlist.

image

Where is alice? :thinking:

**Comment by kesselb** [comment]
1. Login as admin
2. Go to user management
3. Enter editing mode for yourself
4. Try to leave it (seems to work for any other user)

Peek 2019-12-23 21-01

**Review by gary-kim — CHANGES_REQUESTED** [review]
> 1. Login as admin
> 2. Go to user management
> 3. Enter editing mode for yourself
> 4. Try to leave it (seems to work for any other user)

It seems that you currently cannot save when editing your own user.

https://github.com/nextcloud/server/blob/32d279d60754b72e332a148d63272650056b9e98/apps/settings/src/components/UserList/UserRow.vue#L211-L218

**Comment by gary-kim** [comment]
> Where is alice? thinking
> 
> image
> 
> Alice is returned by the server.

https://github.com/nextcloud/server/blob/32d279d60754b72e332a148d63272650056b9e98/apps/settings/src/components/UserList.vue#L311-L314

Seems to be intentional. I think it'd make more sense if their own user showed up but wasn't editable though.

**Comment by rullzer** [comment]
Ping :)

**Comment by rullzer** [comment]
master is NC19 now.
Please backport if it needs to go into 18

**Comment by kesselb** [comment]
Conflicts :fearful:

**Comment by wiswedel** [comment]
🏓

**Comment by wiswedel** [comment]
Anyone still alive fixing this?

**Comment by juliushaertl** [comment]
Yes, I was waiting for https://github.com/nextcloud/server/pull/19533 to be merged before doing another rebase dance.

**Comment by wiswedel** [comment]
> Yes, I was waiting for #19533 to be merged before doing another rebase dance.

#19533 has been merged to master, but 18 backport failed

**Comment by juliushaertl** [comment]
All comments should be addressed, editing the own user is now limited to admin users as it is done in the backend.

Ready for review.

**Comment by kesselb** [comment]
What is that 3rdparty update about @juliushaertl? :confused:

**Comment by juliushaertl** [comment]
Oh seems that got in there by accident, I'll fix that.

**Comment by kesselb** [comment]
If Alice is a subdomain for group Test1234 and access the user list I see other users but not myself. I would expect to at least see Alice even if editing is disable. But not a blocker for this one. It's just confusing.

**Comment by kesselb** [comment]
Peek 2020-02-25 19-58

As group admin (probably also as admin) the form for adding a user the page width is weird.

**Comment by kesselb** [comment]
As group admin add a new user without selecting a group. The warning is shown for the button not the group list. Also not a blocker.

image

**Comment by juliushaertl** [comment]
> If Alice is a subdomain for group Test1234 and access the user list I see other users but not myself. I would expect to at least see Alice even if editing is disable. But not a blocker for this one. It's just confusing.

That actually works for me: user1 is group admin of group1
image

**Comment by kesselb** [comment]
Also anything related to showing quota seems to be broken. It seems to be possible to change the quota but not what quota is currently selected. But that's probably fixed by https://github.com/nextcloud/server/pull/18818.

**Comment by juliushaertl** [comment]
> Also anything related to showing quota seems to be broken. It seems to be possible to change the quota but not what quota is currently selected. But that's probably fixed by #18818.

Oh joy, but yes, let's try to address that in #18818 then.

Same for the others. I can look into those, maybe you could create an issue for them, so I have a list rather than picking up pr comments.

**Comment by kesselb** [comment]
> That actually works for me: user1 is group admin of group1
> image

image

Ah. I made Alice group admin but not member of the group. But yes a different story because Alice is missing in the user response.

**Review by kesselb — APPROVED** [review]

**Review by gary-kim — APPROVED** [review]
Tested and looked at the code. LGTM.

P.S. Never seen a computed returning a function before. Interesting.

**Comment by juliushaertl** [comment]
> P.S. Never seen a computed returning a function before. Interesting.

Yeah the benefit of that is that the result is still cached, so it will only update if the dependency changes in contrast to a method.

**Comment by nursoda** [comment]
I seem to be missing something here: I just installed 18.0.3 and still I can't edit my own user (no pencil icon). I need that e.g. to be able to assign myself to new groups. Currently, my (clumsy) workaround is to impersonate another admin and modify my user there…

users

**Comment by wiswedel** [comment]
> I seem to be missing something here: I just installed 18.0.3 and still I can't edit my own user (no pencil icon). I need that e.g. to be able to assign myself to new groups. Currently, my (clumsy) workaround is to impersonate another admin and modify my user there…

I confirm that.
But even the impersonation trick does not work for the first admin that's been configured at installation. 

image

**Comment by rullzer** [comment]
yeah 18.0.3 is a release with just a security fix.
This will come with 18.0.4

**Comment by vincentstans** [comment]
> > Can i fix it by downloading and replacing some files ?
> 
> Sorry, @Uwe-MrT. It's only gonna be in 18.0.3, unless you check out the source and compile the Vue stuff by yourself. Then you can have it today.

**Comment by kesselb** [comment]
@vincentstans what part of "18.0.3 is a release with just a security fix" did you not understand?

**Comment by Hydrog3n** [comment]
Hello, 
This seems not working in 19.0.0Beta4 with only one user in my case. 

ksnip_20200416-212430

**Comment by nursoda** [comment]
Works fine in 19 beta 5, thanks!

**Pull request merged**